### PR TITLE
Refactored Events to allow for mocking and testing

### PR DESCRIPTION
**Pull request by jclarkin**
- Also tossed in an Event structure test

### CODE REVIEW

File: src/notify/event/event_test.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+func TestEvent(t *testing.T) {
+	var e Event
+	var eJSON = []byte(`
+    {
+        "id": "5b2e1b6f-b3f7-4a94-8b05-aa441606d886",
+        "event_type": "test_event_type",
+        "context": "test_context",
+        "original_account_id": "e4f05126-a0e8-437d-94f9-b553c959cdfb",
+				"payload": "Инстинкт плюс возможность, равно прибыль.",
```
Comment (jclarkin) [inline src/notify/event/event_test.go:19]: Apparently... Toss in some russian, and Github does even worse tabbing!

File: src/notify/jsonUtils/jsonUtils.go [hunk 2]
```diff
@@ -24,6 +24,7 @@ func Output(w http.ResponseWriter, httpStatus int, data interface{}) {
 	w.Write(eJSON)
 }
 
+// Decode will try and parse the content of a Reader into the matchig interface
```
Comment (snocorp) [inline src/notify/jsonUtils/jsonUtils.go]: Typo: matchig

File: src/notify/router/router.go [hunk 3]
```diff
@@ -21,9 +21,10 @@ func Setup() {
 	router.GET("/", index)
 	router.GET("/status", status)
 
-	router.POST("/events", event.Create)
-	router.GET("/events", event.Index)
-	router.GET("/events/:id", event.Show)
+	eHandler := event.NewHandler()
```
Comment (snocorp) [inline src/notify/router/router.go]: evt or event might be better than e

Comment (jclarkin) [inline src/notify/router/router.go]: `evtEvent` ? 😝

File: src/notify/subscription/db.go [hunk 4]
```diff
@@ -2,8 +2,11 @@ package subscription
 
 import "database/sql"
 
-type Persistance struct{}
-type Persist interface {
+// PgDatabase is a structure that implements the Database interface and interacts with Postgres db
+type PgDatabase struct{}
+
+// Database is an interface for persisting a structure
```
Comment (snocorp) [inline src/notify/subscription/db.go]: I would replace 'structure' with 'Subscription' and maybe persisting and retrieving

**Pull request merged**